### ISSUE TITLE
cypress-ntlm launcher does not set HTTPS_PROXY

### ISSUE THREAD

**Issue by plnordquist** [bjowes/cypress-ntlm-auth#101]
# Environment
cypress 4.0.2
cypress-ntlm-auth 2.2.1

# Issue
When executing cypress, the cypress-ntlm launcher does not set the HTTPS_PROXY environment variable so the value from the environment and the value from the npm config can overwrite the ntlm proxy configuration that cypress uses to run tests.

The following shows up in the debug logs when HTTPS_PROXY is set and I run cypress-ntlm run:

```
cypress:server:util:proxy found proxy environment variables { NO_PROXY: '<-loopback>', HTTP_PROXY: 'http://127.0.0.1:54580', HTTPS_PROXY: 'http://<corporate-proxy>', no_proxy: '<-loopback>', http_proxy: 'http://127.0.0.1:54580', https_proxy: 'http://<corporate-proxy>', npm_config_proxy: 'http://<corporate-proxy>', npm_config_https_proxy: 'http://<corporate-proxy>' }
cypress:server:util:proxy overriding uppercase env var with lowercase { name: 'NO_PROXY' }
cypress:server:util:proxy overriding uppercase env var with lowercase { name: 'HTTP_PROXY' }
cypress:server:util:proxy overriding uppercase env var with lowercase { name: 'HTTPS_PROXY' }
cypress:server:util:proxy normalized proxy environment variables { NO_PROXY: '<-loopback>', HTTP_PROXY: 'http://127.0.0.1:54580', HTTPS_PROXY: 'http://<corporate-proxy>' }
```

The following shows up in the debug logs when I specifically unset HTTPS_PROXY and run cypress-ntlm run:

```
cypress:server:util:proxy found proxy environment variables { NO_PROXY: '<-loopback>', HTTP_PROXY: 'http://127.0.0.1:59566', no_proxy: '<-loopback>', http_proxy: 'http://127.0.0.1:59566', npm_config_proxy: 'http://<corporate-proxy>/', npm_config_https_proxy: 'http://<corporate-proxy>/' }
cypress:server:util:proxy overriding uppercase env var with lowercase { name: 'NO_PROXY' }
cypress:server:util:proxy overriding uppercase env var with lowercase { name: 'HTTP_PROXY' }
cypress:server:util:proxy using npm's npm_config_https_proxy as HTTPS_PROXY
cypress:server:util:proxy normalized proxy environment variables { NO_PROXY: '<-loopback>', HTTP_PROXY: 'http://127.0.0.1:59566', HTTPS_PROXY: 'http://<corporate-proxy>/' }
```

I could unset the npm config variables and get this working but I think that could break some post scripts of packages as they are installed.

**Comment by bjowes** [comment]
Hi @plnordquist , thanks for reporting this.
I see that the launcher needs to override HTTPS_PROXY to fix this, will release a new version soon.

**Comment by bjowes** [comment]
Fixed in 2.2.2. 
Please provide feedback if it works for you.

**Comment by plnordquist** [comment]
Hmm the fix actually seemed to break the tests, here's the output from the proxy debug lines

```
cypress:server:util:proxy found proxy environment variables { npm_config_proxy: 'http://<corporate-proxy>/', npm_config_https_proxy: 'http://<corporate-proxy>/' }
cypress:server:util:proxy using npm's npm_config_proxy as HTTP_PROXY
cypress:server:util:proxy using npm's npm_config_https_proxy as HTTPS_PROXY
cypress:server:util:proxy setting default NO_PROXY of ``
cypress:server:util:proxy <-loopback> not found, adding localhost to NO_PROXY
cypress:server:util:proxy normalized proxy environment variables { NO_PROXY: '127.0.0.1,::1,localhost', HTTP_PROXY: 'http://<corporate-proxy>/', HTTPS_PROXY: 'http://<corporate-proxy>/' }
```

And here's the error from the tests:

```
cypress:plugin:ntlm-auth Error: HTTP_PROXY environment variable (http://proxy01.pnl.gov:3128/) is not set to current NTLM proxy url (http://127.0.0.1:65468).
Promise {
  <rejected> Error: HTTP_PROXY environment variable is not set to current NTLM proxy url (http://127.0.0.1:65468). Make sure cypress is started using the cypress-ntlm launcher.
```

I was thinking the launcher could just set HTTPS_PROXY where it was setting HTTP_PROXY and I did some testing locally and it seemed to work.  I've posted a PR for my changes at #103.  I don't understand why process.env seemed to be case insensitive when it was deleting variables but maybe it's that way on Windows?

**Comment by bjowes** [comment]
Ah, you had npm_config_proxy and npm_config_https_proxy set also. Missed that on my first analysis.
Thanks for the PR, but I think it will work for you but not for others. Cypress prefers lowercase environment variables, so if process.env.http_proxy is set it will overwrite process.env.HTTP_PROXY. Hence I think it is safer to simply delete all the lowercase variants. I will include the npm_config* variants and make a new release.

**Comment by bjowes** [comment]
And yes, environment variable names are case insensitive on Windows. Seems like node is trying to mimic that behavior.

**Comment by bjowes** [comment]
Improved the fix in 2.2.3. Hope it works better for you.
